ddl/ddl_api.go

```diff
@@ -1313,7 +1313,7 @@ func setDefaultAndComment(ctx sessionctx.Context, col *table.Column, options []*
 			return errUnsupportedModifyColumn.Gen("unsupported modify column constraint - %v", opt.Tp)
 		case ast.ColumnOptionOnUpdate:
 			// TODO: Support other time functions.
-			if !expression.IsCurrentTimestampExpr(opt.Expr) {
+			if col.Tp != mysql.TypeTimestamp || !expression.IsCurrentTimestampExpr(opt.Expr) {
 				return ErrInvalidOnUpdate.Gen("invalid ON UPDATE for - %s", col.Name)
```

Comment:
Make the error msg be consistent with MySQL. `Invalid ON UPDATE clause for 'c4' column`